examples/plot_anderson_accelerate_gd.py

```diff
@@ -0,0 +1,110 @@
+# Copyright 2021 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+r"""
+Anderson acceleration of gradient descent in ridge regression task.
+===================================================================
+
+For a strictly convex function f, :math:`\nabla_x f(x)=0` implies that :math:`x` is the global optimum :math:`f`.
+
+Consequently the fixed point of :math:`T(x)=x-\eta\nabla_x f(x)` is the optimum of :math:`f`.
+
+NOte that repeated application of the operator :math:`T` coincides exactlty with gradient descent with constant step size :math:`\eta`.
+
+Hence, as any fixed point iteration, Gradient descent can benefit from Anderson acceleration.
+Here, we chose chose :math:`f` as the objective function of ridge regression on some dummy dataset.
+Anderson acceleration reaches the optimal parameters within few iterations, whereas Gradient Descent is slower.
+"""
+
+
+import jax
+import jax.numpy as jnp
+
+from jaxopt import AndersonAcceleration
+from jaxopt import PicardIteration
+
+from jaxopt import objective
+from jaxopt.tree_util import tree_scalar_mul, tree_sub
+
+
+import matplotlib.pyplot as plt
+from sklearn import datasets
+
+
+# retrieve intermediate iterates.
+def run_all(solver, w_init, *args, **kwargs):
+  sol, state = solver.init(w_init, *args, **kwargs)
+  sols, errors = [sol], [state.error]
+  for _ in range(solver.maxiter):
+    sol, state = solver.update(sol, state, *args, **kwargs)
+    sols.append(sol)
+    errors.append(state.error)
+  return jnp.stack(sols, axis=0), errors
+
+"""Simple ridge regression with l2 regularization."""
+def ridge_objective(params, l2reg, data):
+  least_squares = objective.least_squares(params, data)
+  ridge = 0.5 * l2reg * jnp.sum(params ** 2)
+  return least_squares + ridge
```

Comment:
Let's add this to `objective.py` (I didn't realize it wasn't the case yet!). Then it also needs to be listed in `objective_and_loss.rst`.